### PR TITLE
added c680 to detect repeat read names

### PR DESCRIPTION
**Pull request by edsu7**
- added c680 test
- c680 test files : `anon_chr1_sameReadName_diffReadGroup` & `anon_chr1_sameReadName_sameReadGroup`
- updated expected reports

### CODE REVIEW

File: seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+                    self.status = 'INVALID'
+                    return              
+                else:
+                    query_bams[rg['file_r1']]={}
+                    query_bams[rg['file_r1']]['is_paired_end']=rg['is_paired_end']
+                    query_bams[rg['file_r1']]['rg']=[]
+                    query_bams[rg['file_r1']]['rg'].append(rg['read_group_id_in_bam'])
+            elif rg['file_r1'] in query_bams:
+                query_bams[rg['file_r1']]['rg'].append(rg['read_group_id_in_bam'])
+
```
Comment (lindaxiang) [inline seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py:83]: Do we have a check in place to do existence checks for fields like `read_groups` and `is_paired_end`? We can for sure do this kind of house keeping checks in separate checks.

**Review by lindaxiang — CHANGES_REQUESTED** [review]
A few minor changes.

File: seq_tools/validation/c670_rg_is_paired_in_bam.py [hunk 2]
```diff
@@ -27,7 +27,18 @@
...
     def __init__(self, ctx, metadata, skip=False):
-        super().__init__(ctx, metadata, __name__, skip=skip)
+        super().__init__(
+            ctx,
+            metadata=metadata,
+            checker_name=__name__,
+            depends_on=[
+                "c608_bam_sanity",
+                "c610_rg_id_in_bam",
+                "c608_bam_sanity",
```
[the remark this answers is not in the document]
Comment (edsu7) [reply seq_tools/validation/c670_rg_is_paired_in_bam.py:37]: Replaced 'c608' with 'c605'
https://github.com/icgc-argo/seq-tools/blob/19678023562709ee5c26142959b7d4850068b08e/seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py#L36

File: seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata, skip=False):
+        super().__init__(
+            ctx,
+            metadata=metadata,
+            checker_name=__name__,
+            depends_on=[
+                "c608_bam_sanity",
+                "c610_rg_id_in_bam",
+                "c608_bam_sanity",
```
[the remark this answers is not in the document]
Comment (edsu7) [reply seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py]: Replaced 'c608' with 'c605'
https://github.com/icgc-argo/seq-tools/blob/19678023562709ee5c26142959b7d4850068b08e/seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py#L36

File: seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py (hunk 1 above)
Comment (edsu7) [reply seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py:70]: Removed redundant tests now that dependancies are inplace
https://github.com/icgc-argo/seq-tools/blob/19678023562709ee5c26142959b7d4850068b08e/seq_tools/validation/c680_repeated_read_names_per_group_in_bam.py#L61-69

**Review by lindaxiang — APPROVED** [review]
LGTM

**Pull request merged**